packages/core/getUserAvailability.ts

```diff
@@ -520,10 +521,43 @@ const _getBusyTimesFromLimits = async (
     performance.measure(`checking duration limits took $1'`, "durationLimitsStart", "durationLimitsEnd");
   }
 
+  const busyTimes = limitManager.getBusyTimes().sort((a, b) => (dayjs(a.start).isBefore(b.start) ? -1 : 1));
+
+  if (eventType.seatsPerTimeSlot) {
+    for (let i = 0; i < bookings.length; i++) {
+      const booking = bookings[i];
+
+      if (
+        booking.attendeesCount &&
+        booking.attendeesCount < eventType.seatsPerTimeSlot &&
+        dayjs(booking.start).isBetween(dateFrom, dateTo)
+      ) {
+        for (let j = i; j < busyTimes.length; j++) {
+          const busyTime = busyTimes[j];
+          if (dayjs(booking.start).add(1, "ms").isBetween(busyTime.start, busyTime.end)) {
+            busyTimes.splice(
+              j,
+              1,
+              {
+                start: busyTime.start,
+                end: dayjs(booking.start).subtract(1, "ms").toISOString(),
+              },
+              {
+                start: dayjs.utc(booking.end).toISOString(),
+                end: busyTime.end,
+              }
+            );
+            break;
+          }
+        }
+      }
+    }
+  }
```

Comment:
can you add some self-review comments here, I can't follow what we are trying to do here

Reply:
![Explanation](https://github.com/calcom/cal.com/assets/86384652/33f4a2f0-a2ab-4ee8-96e6-8aa023030127)